File: App07_Movie/app/src/main/java/com/callor/movies/SecondFragment.java
package com.callor.movies;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.callor.movies.databinding.FragmentSecondBinding;
import com.callor.movies.service.NaverApiService;
import com.callor.movies.service.impl.NaverMovieServiceImplV1;

public class SecondFragment extends Fragment {

    private FragmentSecondBinding binding;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {

        binding = FragmentSecondBinding.inflate(inflater, container, false);

        if (getArguments() != null){
            Log.d("recieved args", getArguments().toString());

            // 전달받은 매개변수 중에 변수 이름이 movie_search 라는 친구가 있으면
            // 그 값을 movie_search 변수에 담아라
            // 만약 없으면 기본 값인 ""을 변수에 담아라
            String movie_search = getArguments().getString("movie_search","");
            Log.d("movie_search", movie_search);
            NaverApiService naverApiService = new NaverMovieServiceImplV1(binding.movieListView);
            naverApiService.getNaverData(movie_search);
        }
        return binding.getRoot();

    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

}

File: hello/app/src/main/java/com/callor/hello/SecondFragment.kt
package com.callor.hello

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController
import com.callor.hello.databinding.FragmentSecondBinding

/**
 * A simple [Fragment] subclass as the second destination in the navigation.
 */
class SecondFragment : Fragment() {

    private var _binding: FragmentSecondBinding? = null
//    private lateinit var _binding : FragmentSecondBinding

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        _binding = FragmentSecondBinding.inflate(inflater, container, false)
        return binding.root

    }

    /*
    fragment 화면에서는 onCreateView() 에서는 root 를 binding 하고 기타 view Component 의
    event 핸들러 설정은 onViewCreate() 에서 설정한다
     */
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        /*
        첫번째로 돌아가기 버튼을 클릭하면 nav_graph.xml 에 설정된
        action_SecondFragment_to_FirstFragment ID 찾아서 destination 에 설정된
        fragment 를 현재 화면에 올려라
         */
        binding.buttonSecond.setOnClickListener {
            findNavController().navigate(R.id.action_SecondFragment_to_FirstFragment)
        }

        /*
        세번째 화면보기 버튼을 클릭하면 nav_Graph.xml 에 설정된
        action_SecondFragment_to_BlankFragment ID 찾아서 destination 에 설정된
        화면으로 교체하라
         */
        binding.buttonThird.setOnClickListener {
            findNavController().navigate(R.id.action_SecondFragment_to_BlankFragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}